src/Pitbulk/Saml2/Saml2User.php

```diff
@@ -56,6 +45,38 @@ function getAttributes()
         return $this->auth->getAttributes();
     }
 
+    /**
+     * @return array attributes retrieved from assertion, mapped by settings
+     */
+    public function getMappedAttributes()
+    {
+        $unmappedAttributes = $this->getAttributes();
+        if (empty($unmappedAttributes)) {
+            return $unmappedAttributes;
+        }
+        $samlSettings = Config::get('laravel4-saml2::saml_settings');
+        $mappedAttributes = [];
+        if (isset($samlSettings['attrMapping'])) {
+            foreach ($samlSettings['attrMapping'] as $mapped => $unmapped) {
+                $mappedAttributes[$mapped] = $unmappedAttributes[$unmapped];
+            }
+        }
+        return $mappedAttributes;
+    }
+
+    /**
+     * @return string the mapped attribute from the assertions
+     */
+    public function getMappedAttribute($attribute)
+    {
+        $attributes = $this->getMappedAttributes();
+        if (array_key_exists($attribute, $attributes)) {
+            return $attributes[$attribute];
+        }
+        return '';
```

Comment:
I prefer to return null so you can distinguish between when there is an attribute with empty value.

Reply:
Please change that so I can merge.

Reply:
Yes, that is a better solution!